src/readers/OrderReader.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: BUSL-1.1
+// This code is made available under the terms and conditions of the Business Source License 1.1 (BUSL-1.1).
+// The act of publishing this code is driven by the aim to promote transparency and facilitate its utilization for educational purposes.
+
+pragma solidity 0.8.18;
+
+// interfaces
+import { ConfigStorage } from "../storages/ConfigStorage.sol";
+import { PerpStorage } from "../storages/PerpStorage.sol";
+import { LimitTradeHandler } from "../handlers/LimitTradeHandler.sol";
+import { OracleMiddleware } from "../oracles/OracleMiddleware.sol";
+
+import { IConfigStorage } from "../storages/interfaces/IConfigStorage.sol";
+import { IPerpStorage } from "../storages/interfaces/IPerpStorage.sol";
+import { ILimitTradeHandler } from "../handlers/interfaces/ILimitTradeHandler.sol";
+import { IOracleMiddleware } from "../oracles/interfaces/IOracleMiddleware.sol";
+
+contract OrderReader {
+  IConfigStorage public configStorage;
+  ILimitTradeHandler public limitTradeHandler;
+  OracleMiddleware public oracleMiddleware;
+  IPerpStorage public perpStorage;
+
+  constructor(address _configStorage, address _perpStorage, address _oracleMiddleware, address _limitTradeHandler) {
+    configStorage = IConfigStorage(_configStorage);
+    perpStorage = IPerpStorage(_perpStorage);
+    limitTradeHandler = ILimitTradeHandler(_limitTradeHandler);
+    oracleMiddleware = OracleMiddleware(_oracleMiddleware);
+  }
+
+  struct ExecutableOrderVars {
+    ILimitTradeHandler.LimitOrder[] orders;
+    bool[] isInValidMarket;
+    ILimitTradeHandler.LimitOrder[] executableOrders;
+    IConfigStorage.MarketConfig[] marketConfigs;
+    uint256 ordersCount;
+    uint256 startIndex;
+    uint256 endIndex;
+  }
+
+  function getExecutableOrders(
+    uint256 _limit,
+    uint256 _offset,
+    bytes32[] memory _assetIds,
+    uint256[] memory _prices
```

Comment:
do we need uint256[] to fit pricese?
I think it mostly 8 expo, so uint64[] would still be enough